scripts/setup-ros-workspace.bash

```diff
@@ -1,135 +1,309 @@
+  if [ -z "$1" ]; then
+    print_and_exit "No ros_team_ws_file given. Cannot setup workspace defines."
+  fi
+  local ros_team_ws_file=$1
+  if [ -z "$2" ]; then
+    print_and_exit "Not specified if docker should be used. Cannot setup workspace defines."
+  fi
+  local use_docker=$2
+    if [ -z "$2" ]; then
+    print_and_exit "Not specified if ros_teamws_file is for docker. Cannot setup workspace defines."
+  fi
+  local is_docker_rtw_file=$3
+
+  # Create a function name
+  fun_name_suffix=${chosen_ros_distro}
+
+  if [ "$ros_ws_prefix" != "-" ]; then
+    fun_name_suffix=${fun_name_suffix}_${ros_ws_prefix}
+  fi
+  if [ "$ros_ws_suffix" != "-" ]; then
+    fun_name_suffix=${fun_name_suffix}_${ros_ws_suffix}
+  fi
+
+  alias_name=_${ws_full_name}
+  fun_name="RosTeamWS_setup_${fun_name_suffix}"
+
+  cp "$ros_team_ws_file" "$ros_team_ws_file".bkp
+  # Comment out the old configuration if such exists - this is hard if using functions...
+  sed -i -e '/'"$fun_name"'/ s/^#*/OLD_/' "$ros_team_ws_file"
+  sed -i -e '/alias st_ros'"$ros_version=$fun_name"'/ s/^#*/#/' "$ros_team_ws_file"
+
+  if [ "$use_docker" = true ]; then
+    docker_image_tag=$(sed "s/[\/]/-/g" <(echo "ros-team-ws_${ubuntu_version_tag}__${ws_folder}__${ws_full_name}"))
+    docker_host_name="rtw-${ws_full_name}-docker"
+  else
+    docker_image_tag="-"
+    docker_host_name="-"
+  fi
+
+  local docker_support="$use_docker"
+  if [ "$is_docker_rtw_file" = true ]; then
+    source_path_rtw="  source /opt/RosTeamWS/ros_ws_$chosen_ros_distro/src/ros_team_workspace/scripts/environment/setup.bash \"\$RosTeamWS_DISTRO\" \"\$RosTeamWS_WS_FOLDER\" \"\$RosTeamWS_WS_PREFIX\" \"\$RosTeamWS_WS_SUFFIX\""
+    docker_support=false # don't use docker in docker
+  else
+    source_path_rtw="  source $FRAMEWORK_BASE_PATH/ros_ws_\"\$RosTeamWS_DISTRO\"/src/$FRAMEWORK_NAME/scripts/environment/setup.bash \"\$RosTeamWS_DISTRO\" \"\$RosTeamWS_WS_FOLDER\" \"\$RosTeamWS_WS_PREFIX\" \"\$RosTeamWS_WS_SUFFIX\""
+  fi
+
+  echo "" >> "$ros_team_ws_file"
+  echo "$fun_name () {" >> "$ros_team_ws_file"
+  echo "  RosTeamWS_BASE_WS=\"${base_ws}\"" >> "$ros_team_ws_file"
+
+  echo "  RosTeamWS_DISTRO=\"${chosen_ros_distro}\"" >> "$ros_team_ws_file"
+  echo "  RosTeamWS_WS_FOLDER=\"${ws_folder}\"" >> "$ros_team_ws_file"
+  echo "  RosTeamWS_WS_PREFIX=\"${ros_ws_prefix}\"" >> "$ros_team_ws_file"
+  echo "  RosTeamWS_WS_SUFFIX=\"${ros_ws_suffix}\"" >> "$ros_team_ws_file"
+  echo "  RosTeamWS_WS_DOCKER_SUPPORT=\"${docker_support}\"" >> "$ros_team_ws_file"
+  echo "  RosTeamWS_DOCKER_TAG=\"${docker_image_tag}\"" >> "$ros_team_ws_file"
+  echo "$source_path_rtw" >> "$ros_team_ws_file"
+  echo "}" >> "$ros_team_ws_file"
+  echo "alias $alias_name=$fun_name" >> "$ros_team_ws_file"
+
+}
+
+update_config()
+{
+  local rtw_file=~/.ros_team_ws_rc
+
+  setup_ros_team_ws_file "$rtw_file" "$use_docker" "$is_docker_rtw_file"
+
+  # source new workspace
+  source "$rtw_file"
+
+  # Update rosdep definitions
+  rosdep update
```

Comment:
Do we have to call this? I would rather leave this out if possible. The same for the `rospack profile`

Reply:
Do you mean both `source "$rtw_file"` and `rosdep update`?

Reply:
`rosdep update`